hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java

```diff
@@ -289,6 +296,123 @@ private void clear() {
    */
   protected abstract void resetViewState();
 
+  /**
+   * Batch loading all the partitions if needed.
+   *
+   * @return A list of relative partition paths of all partitions.
+   */
+  private List<String> ensureAllPartitionsLoadedCorrectly() {
```

Comment:
left a msg below. we should keep this within MetadataFSV. should not let FS based listing call in these by mistake.

Reply:
After thinking about this again, I think it makes sense to disallow FS-based listing call for all partitions.  Now `getAllPartitionPaths()` throws `HoodieException` if invoked for FS-based listing.